Add Fable Json serialization to Saturn and Giraffe.

isaacabraham opened the pull request:
@theimowski can you have a quick look at what I've done in this PR and see if it looks about right? I don't even know how to build the template locally so this is just being done "blind".

When complete, this will fix #49

theimowski commented:
Looks ok at first glance - you should be able to build using the build scripts in root directory, then add local template with `dotnet new -i nupkg/SAFE.Template.X.X.X.nupkg` (see [Contribution guide](https://github.com/SAFE-Stack/SAFE-template/blob/master/CONTRIBUTING.md#testing-nuget-package))

isaacabraham commented:
OK, I think this is ready for review / test. I tested it by doing the following:

1. Created a new project using the template using `-S saturn`.
2. Changed the type of `Counter` to `{ Value : int }`.
3. Fixed any errors in update:

```fsharp
let update msg (model : Model) =
  let model' =
    match model,  msg with
    | Some x, Increment -> Some { Value = x.Value + 1 }
    | Some x, Decrement -> Some { Value = x.Value - 1 }
    | None, Init (Ok x) -> Some x
    | _ -> None
  model', Cmd.none
```

4. At this point, the app should work. However, if you hit `api/init` directly, the call will fail because it will try to serialize to XML (I assume when Fable makes the client-side call, it requests a JSON response explicitly). So, you can do one of two things:
   a. You can attach the `[<CLIMutable>]` attribute to `Counter` so that it serializes correctly.
   b. You can make the following change in `apiRouter`:

```fsharp
      return! Successful.OK counter next ctx
```
with
```fsharp
      return! json counter next ctx
```

isaacabraham commented on Content/src/Server/paket.references:
```diff
@@ -9,11 +10,15 @@ group Server
-//#elseif (Remoting && Server == "giraffe")
-  Fable.Remoting.Giraffe
-//#elseif (Remoting && Server == "saturn")
+//#elseif (Remoting)
```
This removes a warning during template generation that added the same dependency twice.

theimowski commented:
Is this only applicable for Saturn option?

isaacabraham commented:
Mmm, good point. I suppose it did also be there for Giraffe as well. No clue how this works with Suave though.

isaacabraham commented:
@theimowski now added to Giraffe as well.

isaacabraham commented:
I *think* that this is good to go.

theimowski commented:
cool, thanks! released in 0.11
